sympy/series/limits.py

```diff
@@ -206,13 +206,13 @@ def doit(self, **hints):
             return Order(limit(e.expr, z, z0), *e.args[1:])
 
         try:
-            r = gruntz(e, z, z0, dir)
-            if r is S.NaN:
-                raise PoleError()
-        except (PoleError, ValueError):
             r = heuristics(e, z, z0, dir)
             if r is None:
                 return self
+        except (PoleError, ValueError):
+            r = gruntz(e, z, z0, dir)
```

Comment:
gruntz is what really needs to be in the "try" part of a try-except block. Here its exceptions are not being caught.